app/client/lib/MultiUserManager.ts

```diff
@@ -0,0 +1,182 @@
+import {computed, Computed, dom, DomElementArg, IDisposableOwner, keyframes, Observable, styled} from "grainjs";
+import {cssModalBody, cssModalButtons, cssModalTitle, IModalControl, modal} from 'app/client/ui2018/modals';
+import {bigBasicButton, bigPrimaryButton} from 'app/client/ui2018/buttons';
+import {mediaXSmall, testId, theme, vars} from 'app/client/ui2018/cssVars';
+import {UserManagerModel, IMemberSelectOption} from 'app/client/models/UserManagerModel';
+import {icon} from 'app/client/ui2018/icons';
+import {cssEmailTextarea,} from "app/client/ui/UserItem";
+import {BasicRole, VIEWER, NonGuestRole, isBasicRole} from "app/common/roles";
+import {menu, menuItem} from 'app/client/ui2018/menus';
+
+function parseEmailList(emailListRaw: string): Array<string> {
+  return emailListRaw
+    .split('\n')
+    .map(email => email.trim())
+    .filter(email => email !== "")
+}
+
+function validateEmail(email: string): boolean {
+  const mailformat = /\S+@\S+\.\S+/;
+  return mailformat.test(email);
+}
+
+export function buildMultiUserManagerModal(
+  owner: IDisposableOwner,
+  model: UserManagerModel,
+  onAdd: (email: string, role: NonGuestRole) => void,
+) {
+  const emailListObs = Observable.create(owner, "");
+  const rolesObs = Observable.create<BasicRole>(owner, VIEWER);
+  const isValidObs = Observable.create(owner, true);
+  
+  const enableAdd: Computed<boolean> = computed((use) => Boolean(use(emailListObs) && use(rolesObs) && use(isValidObs)));
+  
+  const save = (ctl: IModalControl) => {
+    const emailList = parseEmailList(emailListObs.get())
+    const role = rolesObs.get();
+    if (emailList.some(email => !validateEmail(email))) {
+      isValidObs.set(false);
+    } else {
+      emailList.map(email => onAdd(email, role))
+      ctl.close()
+    }
+  }
+  const cssBody = model.isPersonal ? cssAccessDetailsBody : cssUserManagerBody;
```

Comment:
```suggestion
  const cssBody = cssUserManagerBody;
```

I believe this shouldn't be necessary since it's only used by `UserManager` for showing access details instead of the full management UI.